internal/native/video.go

```diff
@@ -69,12 +73,32 @@ func (n *Native) VideoSleepModeSupported() bool {
 	return n.sleepModeSupported
 }
 
+// useExtraLock uses the extra lock to execute a function.
+// if the lock is currently held by another goroutine, returns an error.
+//
+// it's used to ensure that only one change is made to the video stream at a time.
+// as the change usually requires to restart video streaming
+// TODO: check video streaming status instead of using a hardcoded timeout
+func (n *Native) useExtraLock(fn func() error) error {
```

Comment:
Could this queuing-up be better handled by pushing the change into a channel to be handled inside a go-routine that serves up the changes in-order?